### PR TITLE
Introduction of gsl::not_null<T*>()

### PR DESCRIPTION
**Pull request by daschuer**
Windows and MacOs will fail until we have merged the vcpkg update.

### CODE REVIEW

File: src/engine/readaheadmanager.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include <QList>
 #include <QPair>
+#include <gsl/gsl>
```
Comment (Swiftb0y) [inline src/engine/readaheadmanager.h:5]:
```suggestion
#include <gsl/pointers>
```

File: packaging/debian/control.in [hunk 2]
```diff
@@ -59,6 +59,7 @@ Build-Depends: debhelper (>= 11),
                libudev-dev,
 # Note: libdjinterop is available within the Mixxx PPA
                libdjinterop-dev,
+               libmsgsl-dev,
```
[the remark this answers is not in the document]
Comment (daschuer) [reply packaging/debian/control.in:62]: It does, as "guidelines-support-library" https://repology.org/project/ms-gsl/versions

[the remark this answers is not in the document]
Comment (uklotzde) [reply packaging/debian/control.in:62]: Development-only: `guidelines-support-library-devel`

[the remark this answers is not in the document]
Comment (Swiftb0y) [reply packaging/debian/control.in:62]: Thanks, I should've looked better.

**Comment by daschuer** [comment]
@Holzhaus how can we add ms-gsl to Arch Linux?

**Comment by Holzhaus** [comment]
> @Holzhaus how can we add ms-gsl to Arch Linux? 

We need to update the docker container and add this: https://archlinux.org/packages/community/any/microsoft-gsl/

We need to do that anyway for Qt 6.3.0.

**Comment by daschuer** [comment]
I do not know how to do it. Can you take care?

**Comment by Holzhaus** [comment]
> I do not know how to do it. Can you take care?

https://github.com/Holzhaus/mixxx/tree/bump-archlinux-qt6-to-6.3.0

**Comment by Swiftb0y** [comment]
@daschuer please rebase. Thanks.

**Review by Swiftb0y — APPROVED** [review]
Thanks. LGTM

**Pull request merged**